File: kyw0716/level 1/과일 장수.js
function solution(k, m, score) {
  const box = [];
  let answer = 0;

  score.sort((a, b) => a - b);

  while (score.length > 0) {
    box.push(score.pop());
    if (box.length === m) {
      const minScore = Math.min(...box);

      answer += m * minScore;

      box.splice(0, box.length);
    }
  }

  return answer;
}

// score를 크기순으로 정렬한다.
// 배열에 m만큼 사과를 담는다.
// 배열 내 최솟값을 구한다.
// m * 최솟값을 answer에 더한다.
// 위과정을 반복한다.

//https://school.programmers.co.kr/learn/courses/30/lessons/135808

File: sk000801/LEVEL1/과일 장수.java
import java.util.*;
class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        int index = 0;
        Arrays.sort(score);
        int[] scoreMax = new int[score.length];
        for (int i = 0; i < score.length; i++) {
            scoreMax[i] = score[score.length-i-1];
        }
        for (int i = 1; i <= score.length/m; i++) {
            answer += scoreMax[i*m-1]*m;
        }
        return answer;
    }
}